umbra-js/src/classes/Umbra.ts

```diff
@@ -313,107 +313,161 @@ export class Umbra {
-          if (isForUser) userAnns.push({ randomNumber, receiver, amount, token, from, txHash, timestamp, isWithdrawn });
-          return userAnns;
-        }, [] as UserAnnouncement[]);
-
-        // Filtering and parsing done, return announcements
-        return { userAnnouncements };
+        // Map the subgraph amount field from string to BigNumber
+        const subgraphAnnouncements = await this.fetchAllAnnouncementsFromSubgraph(startBlock, endBlock);
+        return subgraphAnnouncements.map((x) => ({ ...x, amount: BigNumber.from(x.amount) }));
       } catch (err) {
-        // Graph query failed, try requesting logs directly IF we are not on polygon. If we are on
-        // Polygon, there isn't much we can do, so for now just show a warning and return an empty array
-        if (this.chainConfig.chainId === 137) {
-          console.warn('Cannot fetch Announcements from logs on Polygon, please try again later');
-          return { userAnnouncements: [] };
-        }
-        const userAnnouncements = await this.userAnnouncementsFromLogs(spendingPublicKey, viewingPrivateKey, startBlock, endBlock); // prettier-ignore
-        return { userAnnouncements };
+        return this.fetchAllAnnouncementFromLogs(startBlock, endBlock);
       }
     }
 
-    // Subgraph not available, try requesting logs directly IF we are not on Polygon. If we are on
-    // Polygon, there isn't much we can do, so for now just show a warning and return an empty array
-    if (this.chainConfig.chainId === 137) {
-      console.warn('Cannot fetch Announcements from logs on Polygon, please try again later');
-      return { userAnnouncements: [] };
-    }
-    const userAnnouncements = await this.userAnnouncementsFromLogs(spendingPublicKey, viewingPrivateKey, startBlock, endBlock); // prettier-ignore
-    return { userAnnouncements };
+    return this.fetchAllAnnouncementFromLogs(startBlock, endBlock);
   }
 
-  // ======================================= HELPER METHODS ========================================
+  /**
+   * @notice Fetches all Umbra event logs using The Graph
+   * @dev Currently ignores the start and end block parameters and returns all events; this may change in a
+   * future version
+   * @param startBlock Ignored
+   * @param endBlock Ignored
+   * @returns A list of Announcement events supplemented with additional metadata, such as the sender, block,
+   * timestamp, txhash, and the subgraph identifier
+   */
+  async fetchAllAnnouncementsFromSubgraph(
+    startBlock: string | number,
+    endBlock: string | number
+  ): Promise<SubgraphAnnouncement[]> {
+    if (!this.chainConfig.subgraphUrl) {
+      throw new Error('Subgraph URL must be defined to fetch via subgraph');
+    }
+
+    // TODO: We're ignoring these overrides for The Graph. Is this intentional? Should we remove the parameters,
+    // or update so it uses them?
+    startBlock;
+    endBlock;
```

Comment:
ah these were being ignored originally before the refactor, right? I think that's just an oversight on my part when implementing the subgraph query

we probably should respect those values to save the user scanning time, but it's advanced-mode only so likely not a common path, so I'm ok punting on it and tracking it in an issue (unless you want to change it in this PR, then of course feel free)

Reply:
Yep these were ignored prior to the refactor, it just became more obvious when they were split into stand alone methods. I created an issue to track here: https://github.com/ScopeLift/umbra-protocol/issues/275